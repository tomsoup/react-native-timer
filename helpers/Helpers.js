/**
 * Format second
 * @param  {string} totalSeconds
 * @return {string}  New board
 */
export const formatSeconds = (totalSeconds) => {
  let seconds = totalSeconds % 60;
  let minutes = Math.floor(totalSeconds / 60);

  if (seconds < 10) {
    seconds = `0${seconds}`
  }
  if (minutes < 10) {
    minutes = `0${minutes}`
  }
  return `${minutes}:${seconds}`;
}
