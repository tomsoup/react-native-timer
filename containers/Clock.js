import React, { Component }  from 'react';
import { StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { formatSeconds } from '../helpers/Helpers';

export const Clock = ({ totalSeconds = 0}) => {

    return (
      <View style={{
        backgroundColor: 'rgba(8, 214, 160, 0.5)',
        paddingHorizontal: 60,
        paddingVertical: 85,
        borderRadius: 200,
        borderWidth: 3,
        borderColor: '#92b5ac'
      }}>
        <Text style={{
          color: 'white',
          fontSize: 34
        }}>
          {formatSeconds(totalSeconds)}
        </Text>
      </View>
    );
}


Clock.propTypes = {
  totalSeconds: PropTypes.number
};
