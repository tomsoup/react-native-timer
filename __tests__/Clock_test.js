import React from 'react';
import 'react-native';
import { formatSeconds } from '../helpers/Helpers';
import Clock from '../containers/Clock';
import renderer from 'react-test-renderer';


describe("Clock", () => {
  it('render correctly', () => {
    const clock = renderer.create(
    <Clock />
    ).toJSON();
    expect(clock).toMatchSnapshot();
  });

  describe("formating seconds", () => {
    it('should format new seconds', () => {
      const seconds = 615;
      const expected = '10:15';
      const actual = formatSeconds(seconds);
      expect(actual).toBe(expected);
    });

    it('should format new seconds min/sec before 13:00', () => {
      const seconds = 61;
      const expected = '01:01';
      const actual = formatSeconds(seconds);
      expect(actual).toBe(expected);
    });
  })
})
