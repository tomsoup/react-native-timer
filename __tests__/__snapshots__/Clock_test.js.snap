// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock render correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Clock
  </Text>
</View>
`;

exports[`Clock render correctly 2`] = `[Function]`;
